### PR TITLE
DCV-2572 dbt-coves blue-green

### PR DESCRIPTION
**Pull request by BAntonellini**
This PR implements `dbt-coves blue-green` command

It's basically a reimplementation of Balboa's [blue_green_run script](https://github.com/datacoves/balboa/blob/main/automate/blue_green_run.py), with the main difference here we use pure Python instead of `Python + dbt macros`

**WARNING**: use testing databases like `BRUNO_TEST`, as data can be lost during swap/drop.

A dbt blue-green run consists of copying a blue (production) database into a green (staging one), making sure dbt builds correctly, swapping them and dropping the old production one.

This process runs the following:
- Create staging database
- Clone schemas and schema-grants from prod to staging
- Run dbt build
- Clone database grants from prod to staging
- Swap prod and staging (staging becomes prod and viceversa)
- Drop staging database (ex production)

### CODE REVIEW

File: dbt_coves/tasks/blue_green/clone_db.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        for v in self.thread_commands:
+            proc = threading.Thread(target=self.run_commands, args=(v,))
+            procs.append(proc)
+            proc.start()
+        # complete the processes
+        for proc in procs:
+            proc.join()
+
+
+# if __name__ == "__main__":
```
Comment (ssassi) [inline dbt_coves/tasks/blue_green/clone_db.py:191]: Could we avoid keeping this commented out and test this differently?

Comment (BAntonellini) [reply dbt_coves/tasks/blue_green/clone_db.py:191]: @ssassi I agree, it belongs to the code Coldbore sent us, and its testing case. I'd like to discuss how all this should be tested.

File: dbt_coves/tasks/blue_green/main.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    def _check_and_drop_staging_db(self):
+        """
+        Checks if the staging database exists and drops it if it does.
+
+        Returns:
+            None
+        """
+        green_exists = self._check_if_database_exists()
+        if green_exists and self.drop_staging_db:
+            if self.drop_staging_db_after:
```
Comment (ssassi) [inline dbt_coves/tasks/blue_green/main.py:176]: it should also validate that it's a valid int?

File: dbt_coves/tasks/blue_green/main.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+                console.print(
+                    f"Green database {self.staging_database} exists."
+                    f"Waiting {self.drop_staging_db_after} minutes before dropping it"
+                )
+                for i in range(self.drop_staging_db_after):
```
Comment (ssassi) [inline dbt_coves/tasks/blue_green/main.py:181]: Shouldn't this raise an exception if the number of loops ended and the database still exists?

Comment (BAntonellini) [reply dbt_coves/tasks/blue_green/main.py:181]: Agree, if you explicitly set it to drop, and give it X minutes to try and drop, something might've happened and blue-green cannot continue

Done.

**Pull request merged**